### ISSUE TITLE
GaussianCriterion gradient issue

### ISSUE THREAD

**Issue by syyeung** [y0ast/VAE-Torch#13]
In GaussianCriterion.lua the gradient with respect to log(sigma^2) is computed as:

`self.gradInput[2] = torch.exp(-input[2]):cmul(torch.add(target,-1,input[1]):pow(2)):mul(-1):add(0.5)
`

But it seems to me that the multiplication should be by -0.5, not -1, such that 

`self.gradInput[2] = torch.exp(-input[2]):cmul(torch.add(target,-1,input[1]):pow(2)):mul(-0.5):add(0.5)`

which would also be consistent with the expression in your comment (after flipping the sign for neg. log likelihood): `- 0.5 + 0.5 * (x - mu)^2 / sigma^2`

I am seeing a gradient check fail with the original code but it passes after the change.

**Comment by y0ast** [comment]
You are absolutely right, I have no idea why the comment and code were inconsistent. I also flipped the sign of all the equations in the comments, to be fully consistent.

Thanks!
